micropipenv.py

```diff
@@ -720,12 +720,16 @@ def _poetry2pipfile_lock(
                 if dependency_info.get("optional", False):
                     extra_dependencies.add(dependency_name)
 
-                # If the dependency has some markers and it's not specified
-                # as a direct dependency, we should move the markers to the
-                # main dependency definition.
-                if "markers" in dependency_info:
-                    if dependency_name not in pyproject_poetry_section["dependencies"]:
-                        additional_markers[normalize_package_name(dependency_name)].append(dependency_info["markers"])
+            # If the dependency is not direct and has some markers,
+            # we should move the markers to the main dependency definition.
+            # If there are no additional markers, we have to have a record
+            # that the dependency has to be installed unconditionaly and that
+            # we have to skip all other additional markers.
+            if dependency_name not in pyproject_poetry_section["dependencies"]:
+                if isinstance(dependency_info, dict) and "markers" in dependency_info:
+                    additional_markers[normalize_package_name(dependency_name)].append(dependency_info["markers"])
+                else:
+                    additional_markers[normalize_package_name(dependency_name)].append("skip-all-markers")
```

Comment:
Thanks for the fix. The approach looks correct to me. I'm wondering if `skip-all-markers` could be replaced with some language-defined marker that could signalize no markers and would not be a valid marker - e.g. `None` (not to introduce a special value here). I see the `None` is handled specifically down below.

Reply:
I see. My first intention was to use True for that but neither True nor None indicates to me that this value should cause something special. For me, None is like a missing marker that should itself be skipped but without any impact on other markers. And it's used that way below when a package itself has no markers except the additional ones. We can use something like "*" to indicate that the package should always be installed but it's even worse than the self-describing value in the current implementation. I'm out of ideas. Do you have some?

Reply:
Maybe use a named object, like:

```python
skip_all_markers = object()
[...]
additional_markers[normalize_package_name(dependency_name)].append(skip_all_markers)
[...]
if skip_all_markers in markers:
```

Reply:
A special sentinel value like this one might make sense here, thank you. Implemented.